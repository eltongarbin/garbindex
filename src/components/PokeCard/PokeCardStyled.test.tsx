import React from 'react';
import { render } from '@testing-library/react';
import { IconButton } from '@material-ui/core';
import { Visibility as VisibilityIcon } from '@material-ui/icons';

import {
  CardHeaderStyled,
  CardMediaStyled,
  CardActionsStyled,
} from './PokeCardStyled';

describe('PokeCard styled components', () => {
  test('CardHeaderStyled renders correctly', () => {
    const { container } = render(
      <CardHeaderStyled title="test title" subheader="test suheader" />
    );
    expect(container).toMatchSnapshot();
  });

  test('CardMediaStyled renders correctly', () => {
    const { container } = render(
      <CardMediaStyled image="https://image.com" title="test title" />
    );
    expect(container).toMatchSnapshot();
  });

  test('CardActionsStyled renders correctly', () => {
    const { container } = render(
      <CardActionsStyled>
        <IconButton aria-label="See more" onClick={() => {}}>
          <VisibilityIcon />
        </IconButton>
      </CardActionsStyled>
    );
    expect(container).toMatchSnapshot();
  });
});
