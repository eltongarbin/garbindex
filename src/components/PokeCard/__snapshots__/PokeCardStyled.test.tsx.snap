// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeCard styled components CardActionsStyled renders correctly 1`] = `
<div>
  <div
    class="MuiCardActions-root sc-AxiKw VavWC MuiCardActions-spacing"
  >
    <button
      aria-label="See more"
      class="MuiButtonBase-root MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`PokeCard styled components CardHeaderStyled renders correctly 1`] = `
<div>
  <div
    class="MuiCardHeader-root sc-AxjAm iZCNfh"
  >
    <div
      class="MuiCardHeader-content"
    >
      <span
        class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        test title
      </span>
      <span
        class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        test suheader
      </span>
    </div>
  </div>
</div>
`;

exports[`PokeCard styled components CardMediaStyled renders correctly 1`] = `
<div>
  <div
    class="MuiCardMedia-root sc-AxirZ jscETi"
    style="background-image: url(https://image.com);"
    title="test title"
  />
</div>
`;
